src/tests/test_rng.cpp

```diff
@@ -135,11 +191,209 @@ class HMAC_DRBG_Reseed_Tests : public Test
+         if(!mac)
+            {
+            result.note_missing("HMAC(SHA-256)");
+            return result;
+            }
+
+         // make sure the nonce has at least 1/2*security_strength bits
+
+         // SHA-256 -> 128 bits security strength
+         for( auto nonce_size : { 0, 4, 15, 16, 17, 32 } )
+            {
+            if(!mac)
+               {
+               mac = Botan::MessageAuthenticationCode::create("HMAC(SHA-256)");
+               }
+
+            Botan::HMAC_DRBG rng(std::move(mac));
+            result.test_eq("not seeded", rng.is_seeded(), false);
+            std::vector<Botan::byte> nonce(nonce_size);
+            rng.initialize_with(nonce.data(), nonce.size());
+
+            if(nonce_size < 16)
+               {
+               result.test_eq("not seeded", rng.is_seeded(), false);
+               result.test_throws("invalid nonce size", [&rng, &nonce] () { rng.random_vec(16); });
+               }
+            else
+               {
+               result.test_eq("is seeded", rng.is_seeded(), true);
+               rng.random_vec(16);
+               }
+            }
+
+         return result;
+         }
+
+      Test::Result test_prediction_resistance()
+         {
+         Test::Result result("HMAC_DRBG Prediction Resistance");
+
+         auto mac = Botan::MessageAuthenticationCode::create("HMAC(SHA-256)");
+         if(!mac)
+            {
+            result.note_missing("HMAC(SHA-256)");
+            return result;
+            }
+
+         // set max_output_before_reseed = 1, forcing a reseed on every request
+         Request_Counting_RNG counting_rng;
+         Botan::HMAC_DRBG rng(std::move(mac), counting_rng, 1);
+
+         rng.random_vec(16);
+         result.test_eq("first request", counting_rng.randomize_count(), size_t(1));
+
+         rng.random_vec(16);
+         result.test_eq("second request", counting_rng.randomize_count(), size_t(2));
+
+         rng.random_vec(16);
+         result.test_eq("third request", counting_rng.randomize_count(), size_t(3));
+
+         return result;
+         }
+
+      Test::Result test_fork_safety()
+         {
+         Test::Result result("HMAC_DRBG Fork Safety");
+
+#if defined(BOTAN_TARGET_OS_TYPE_IS_UNIX)
+         auto mac = Botan::MessageAuthenticationCode::create("HMAC(SHA-256)");
+         if(!mac)
+            {
+            result.note_missing("HMAC(SHA-256)");
+            return result;
+            }
+
+         // make sure rng is reseeded after every fork
+         Request_Counting_RNG counting_rng;
+         Botan::HMAC_DRBG rng(std::move(mac), counting_rng, 16384);
+
+         rng.random_vec(16);
+         result.test_eq("first request", counting_rng.randomize_count(), size_t(1));
+
+         // fork and request from parent and child, both should output different sequences
+         size_t count = counting_rng.randomize_count();
+         Botan::secure_vector<byte> parent_bytes, child_bytes;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define each identifier in a dedicated statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=cpp%3ASingleDeclarationPerStatement)